### PR TITLE
Fix recently added AndroidMessageHandler test

### PR DESCRIPTION
**Pull request by grendello**
Context: https://github.com/xamarin/xamarin-android/pull/7785
Context: https://github.com/xamarin/xamarin-android/pull/7785/commits/7b2e1728295d476f23d832e48640082c9f424aa0

7b2e1728 added changes to retry the test up to 5 times should
https://httpbin.org fail the request (e.g. because of throttling).
However, I skipped building the changes locally believing that should
anything be wrong, that CI would catch it.  And it did, but the failures
weren't in the `[Tests]` tab but rather in a stage build log: the
`Mono.Android.NET-Tests.csproj` failed to build because the `[Retry]`
attribute is not available in `NUnitLite` which this test suite uses.
Additionally, `Assert.Warn` isn't supported either and the
`DoCompression` method should have been marked `async`.

Murphy must be laughing **really** hard now... :)

It should work now.

### CODE REVIEW

File: tests/Mono.Android-Tests/Xamarin.Android.Net/AndroidMessageHandlerTests.cs [hunk 1]
```diff
@@ -39,20 +39,29 @@ protected override HttpMessageHandler CreateHandler ()
...
+					if (await DoDecompression (urlPath, encoding, jsonFieldName)) {
+						return;
+					}
+					count++;
+				}
 			} catch (Exception ex) {
-				Assert.Warn ("Unexpected exception thrown");
-				Assert.Warn (ex.ToString ());
+				Console.WriteLine ("Unexpected exception thrown");
+				Console.WriteLine (ex.ToString ());
```
Comment (jonpryor) [inline tests/Mono.Android-Tests/Xamarin.Android.Net/AndroidMessageHandlerTests.cs:59]: Should this instead use `TestContext.Out.WriteLine()`?

@dellis1972, @jonathanpeppers?

Comment (jonathanpeppers) [reply tests/Mono.Android-Tests/Xamarin.Android.Net/AndroidMessageHandlerTests.cs:59]: Yes, I think that is the only way it would make it into a log.

Comment (grendello) [reply tests/Mono.Android-Tests/Xamarin.Android.Net/AndroidMessageHandlerTests.cs:59]: neither `TestContext.Out.WriteLine` nor `TestContext.WriteLine` work. It appears that NUnitLite doesn't have that feature either :(

Comment (jonathanpeppers) [reply tests/Mono.Android-Tests/Xamarin.Android.Net/AndroidMessageHandlerTests.cs:59]: Ok here: https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=7395571&view=artifacts&pathAsName=false&type=publishedArtifacts

run-Mono.Android.NET_Tests-Debug.binlog has lines like:
```
02-28 17:33:35.485  7425  7445 I NUnit   : JniValueMarshalerContractTests`1.CreateObjectReferenceArgumentState_DefaultValue 
02-28 17:33:35.486  7425  7445 I NUnit   :     Passed
02-28 17:33:35.486  7425  7445 I NUnit   : JniValueMarshalerContractTests`1.CreateReturnValueFromManagedExpression 
02-28 17:33:35.494  7425  7445 I DOTNET  : # jonp: expected: JniValueMarshaler_String_ContractTests
```
Ok, for this test Console.WriteLine() will work fine:
https://github.com/xamarin/java.interop/blob/77800dda83c2db4d90b501c00069abc9880caaeb/tests/Java.Interop-Tests/Java.Interop/JniValueMarshalerContractTests.cs#L267

It wouldn't work in our MSBuild tests, but Device Tests w/ NUnitLite this is the only option.

**Review by jonathanpeppers — APPROVED** [review]

**Review by jpobst — APPROVED** [review]

**Pull request merged**